api/exec/compose_stack.go

```diff
@@ -115,3 +123,129 @@ func createEnvFile(stack *portainer.Stack) (string, error) {
 
 	return "stack.env", nil
 }
+
+func createNetworkEnvFile(stack *portainer.Stack) error {
+	ss := NewStringSet()
+
+	for _, filePath := range stackutils.GetStackFilePaths(stack) {
+		ssName, err := extractNetworkName(filePath)
+		if err != nil {
+			return err
+		}
+
+		if ssName == nil || ssName.Len() == 0 {
+			continue
+		}
+
+		ss.Merge(ssName)
+	}
+
+	for _, s := range ss.List() {
+		if _, ok := os.LookupEnv(s); ok {
+			ss.Remove(s)
```

Comment:
I'm not 100% sure how network names became env vars.

Reply:
It is scanning yaml file and found network section. Like the following,
version: '3.6'
services:
  nginx-example:
    image: nginx:latest
networks:
  default:
    name: ${test}
    driver: bridge